src/helpers/template-helper.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Software Name : eclipse-iofog/Controller
+ * Version: 2.0.x
+ * SPDX-FileCopyrightText: Copyright (c) 2020-2020 Orange
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * This software is distributed under the <license-name>,
+ * the text of which is available at http://www.eclipse.org/legal/epl-2.0
+ * or see the "license.txt" file for more details.
+ *
+ * Author: Franck Roudet
+ */
+// ninja2 like template engine
+const { Liquid } = require('liquidjs')
+const templateEngine = new Liquid()
+
+/**
+  * Object in depth traversal and right value templateEngine rendering
+  * @param {*} subjects
+  * @param {*} templateContext
+  */
+const rvaluesVarSubstition = async (subjects, templateContext) => {
+  Object.keys(subjects).forEach((key) => {
+    if (typeof subjects[key] === 'object') {
+      rvaluesVarSubstition(subjects[key], templateContext)
+    } else if (typeof subjects[key] === 'string') {
+      subjects[key] = templateEngine.parseAndRenderSync(subjects[key], templateContext)

```

Comment:
Does `parseAndRenderSync` handle circular references? Like what would happen if someone submits this template:
```
apiVersion: iofog.org/v2
kind: Application
metadata:
  name: "{{ self.metadata.name  }}"
spec:
...
```

Reply:
not circular in fact: the value of the key is going to be replace by itself, and no more variable expansion is going to be done further so the result is:`{{ self.metadata.name  }}`